skfp/fingerprints/map.py

```diff
@@ -48,10 +49,12 @@ class MAPFingerprint(BaseFingerprintTransformer):
         Number of iterations performed, i.e. maximum radius of resulting subgraphs.
         Another common notation uses diameter, therefore MAP4 has radius 2.
 
-    variant : {"raw_hashes", "bit", "count"}, default="bit"
-        Which variant to fingerprint to use. ``"raw_hashes"`` returns MinHash values.
-        ``"bit"`` folds values into binary vector, and ``"count"`` uses folding with
-        counting.
+    include_duplicated_shingles : bool, default=False
+        Whether to include duplicated shingles in the final fingerprint.
```

Comment:
With folding and count variant, is this necessary? I mean, doesn't `count` argument already cover this?

Reply:
The two operations are distinct: in the `count` case, you count all times a given bit in the bitvec is extracted by the hash. In the `include_duplicated_shingles`, you count as distinct features that you encounter the same shingle several times.

Reply:
So for the bit version there should be no difference, right? Since duplicate shingles should get hashed into the same index anyway. Count variant will have the difference, since:
- for `include_duplicated_shingles=False`, we only count hashing collisions between different subgraphs
- for `include_duplicated_shingles=True`, we will also include the same subgraph occurring multiple times

Am I right? If that's the case, I think we should just always include duplicates for `count`, and ignore this for `bit`. It just sounds very weird for me that someone would like to count bit collisions, but not same thing being present many times.

Reply:
No, the `include_duplicated_shingles` makes it so that if you encounter X for the nth time you convert it into "X|nth", which makes all of them distinct.

Reply:
This is as it was defined in the original impl btw, I am by no means saying it is something to keep.

Reply:
Ok, so your implementation is definitely correct, then. But I'm not sure if this is something really useful? Unless you see some immediate application, I would just remove this and do as I outlined above for simplicity. We don't have to follow the original implementation 100%, after all.

Reply:
It can conceivably have a better representation to distinguish some molecules that are characterized by strongly repeating patterns, it's something likely worth testing. That being said, I don't have a use case at hand where I could easily test this.

Reply:
Fair point, I'll test it on some polymers. So leave it for now then, and I'm approving the PR